plugin/trino-memory/src/main/java/io/trino/plugin/memory/MemorySplitManager.java

```diff
@@ -72,6 +77,45 @@ public ConnectorSplitSource getSplits(
                 splits.add(new MemorySplit(table.getId(), i, splitsPerNode, dataFragment.getHostAddress(), rows, OptionalLong.empty()));
             }
         }
-        return new FixedSplitSource(splits.build());
+
+        ConnectorSplitSource splitSource = new FixedSplitSource(splits.build());
+        if (enableLazyDynamicFiltering) {
+            splitSource = new DelayedSplitSource(dynamicFilter.isBlocked(), splitSource);
```

Comment:
`dynamicFilter.isBlocked()` may provide a future to wait on multiple times, so we need to keep waiting on it until `dynamicFilter.isAwaitable` is also true. We can do something similar to `io.trino.plugin.kudu.KuduSplitManager#whenCompleted`

Reply:
Good catch